File: courier/read_email_list.py
# read_email_list.py

import csv
import os
from typing import List, Dict

SENSITIVE_DATA_DIR = '../sensitive_data'


def get_contacts_from_csv(file_name: str = 'email_list.csv') -> List[Dict[str, str]]:
    contact_list = []
    file_path = os.path.join(SENSITIVE_DATA_DIR, file_name)
    try:
        with open(file_path, mode='r', newline='') as file:
            reader = csv.DictReader(file)
            for row in reader:
                email = row.get('email', '').strip()
                first_name = row.get('first_name', '').strip()
                contact_list.append({
                    "email": email,
                    "first_name": first_name
                })
    except FileNotFoundError:
        print(f"Error: The file {file_path} was not found.")
    except Exception as e:
        print(f"An error occurred: {e}")
    return contact_list

File: courier/read_email_list.js
const fs = require('fs');
const path = require('path');
const csv = require('csv-parser');

const SENSITIVE_DATA_DIR = path.join(__dirname, 'sensitive_data');

/**
 * Reads contact information (email, first_name) from a CSV file.
 *
 * @param {string} file_name - The name of the CSV file (default: 'email_list.csv').
 * @returns {Promise<Array<{email: string, first_name: string}>>} - A Promise that resolves to an array of contacts.
 */
function getContactsFromCsv(file_name = 'email_list.csv') {
  return new Promise((resolve, reject) => {
    const contacts = [];
    const filePath = path.join(SENSITIVE_DATA_DIR, file_name);

    fs.createReadStream(filePath)
      .pipe(csv())
      .on('data', (row) => {
        contacts.push({
          email: row.email,
          first_name: row.first_name,
        });
      })
      .on('end', () => {
        resolve(contacts);
      })
      .on('error', (err) => {
        reject(err);
      });
  });
}

module.exports = { getContactsFromCsv };
